### PR TITLE
Adds space carp traitor item and carp plushie "attack"

### PR DESCRIPTION
**Pull request by ghost**
Adds a "dehydrated" space carp to the traitor PDA uplink for 7 telecrystals.
It looks and acts like a carp plushie until it comes into contact with water and then animates, becoming a fully fledged space carp ready to gnash anyone.

Animation images:
plushie to carp3

Carp plushie (and also the dehydrated carp) tweaked to provide amusing attack verbs without causing any real harm: eaten, bitten and fin slapped. I felt the toy needed some love beyond existing.

Edit: Hope I did what you said!

dehy_carp is now a child of carpplushie, I even got some code wrong due to trying to duplicate it, that problem shouldn't happen again.

Click the dehydrated carp in your hand to make its faction yours, all it took was looking into three separate files for the staff of animation to understand what the heck I was meant to do.

It now costs 3 TCs.

Oh and I fixed the pathing for you goof, I somehow skipped looking for the basic code before looking for an example in the tgstation code, I took that from the Clown horn if you wanted to check in on that and fix it if it's bad.

Edit2: In the 075328f merge I updated my code in one file and that was the uplink_item.dm in datums so that my code didn't conflict with it, placing the fake plushie underneath the pizzabox bomb.

Edit3: Travis if this doesn't work, I can offer many bribes of assorted flavours.

Edit4: I think I've finished this now.

### CODE REVIEW

File: code/game/objects/items/dehy_carp.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	if(istype(O,/obj/structure/sink))
+		user << "<span class='notice'>You place [src] under a stream of water...</span>"
+		user.drop_item()
+		loc = get_turf(O)
+		return Swell()
+	..()
+
+/obj/item/dehy_carp/proc/Swell()
+	icon = 'icons/mob/animal.dmi'
+	icon_state = "carp_swell"
```
Comment (Iamgoofball) [inline code/game/objects/items/dehy_carp.dm]: can't you just flick() this icon state here?

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1038,6 +1038,17 @@ obj/item/toy/cards/deck/syndicate
 	icon = 'icons/obj/toy.dmi'
 	icon_state = "carpplushie"
 	w_class = 2.0
+	var/bitesound = 'sound/weapons/bite.ogg'
+
+	// Attack mob
+	/obj/item/toy/carpplushie/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
```
Comment (Iamgoofball) [inline code/game/objects/items/toys.dm]: somehow relative pathing and absolute pathing at the same time

i don't know what to say about this

**Comment by Iamgoofball** [comment]
overall :+1:

**Comment by RemieRichards** [comment]
Perhaps make it a subtype of the plushie, so not all plushies get turned into carp

**Comment by CosmicScientist** [comment]
Alright I'll make it a subtype, to be honest I only realised that in between making the pull request and reading that comment.

I'll re-examine what I did with that pathing, I may have been looking at some bad code when looking for examples similar to what I wanted to do elsewhere in the TG Station master code.

I'll look into flick(). Being new to the coding shenanigans of Dream Maker probably meant I didn't know what I was doing.

**Comment by Iamgoofball** [comment]
Wait, no, reopen this

You can just push more commits @CosmicScientist and it autoupdates the PR

**Comment by RemieRichards** [comment]
@CosmicScientist no need to close the PR, you can push new commits to the same PR branch and it updates.

**Comment by CosmicScientist** [comment]
Oh okay, I didn't realise. Thanks!

New to git everything, really concerned about every button I press, spent a lot of hours asking a friend who knows a lot more about this and command line interfaces than me and now it's four in the morning. Time flies when you're trying to argue with Git.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm [hunk 3]
```diff
@@ -272,6 +272,13 @@ datum/reagent/water/reaction_obj(var/obj/O, var/volume)
 			cube.Expand()
 	return
 
+datum/reagent/water/reaction_obj(var/obj/O, var/volume)
+	src = null
+	if(istype(O,/obj/item/dehy_carp))
+		var/obj/item/dehy_carp/dehy = O
+		dehy.Swell() // Makes a carp
+		return
```
Comment (Incoming5643) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm]: There's gotta be a less intrusive way to do this. See how monkey cubes handle it, as this is effectively the same item except with a carp.

**Comment by Incoming5643** [comment]
Fun idea, but should probably be cheaper if the carp isn't friendly to you. You can get equally pissed off carp for cheaper and in larger numbers with an emag.

**Comment by WJohn** [comment]
A space carp has 30 health, I don't think anyone will ever use it again when they realize they can get way the hell more out of anything else.

**Comment by JJRcop** [comment]
@CosmicScientist No no no, you need to hard reset the branch into the detached HEAD before the rebase triggers the pull after the SHA timestamp expires!

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm (hunk 3 above)
Comment (CosmicScientist) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm]: As I'm only just breaking my way into understanding how to code in this, I'm not entirely sure how. To be honest that is effectively the same code as for the monkey cubes minus a check for whether or not the cube is still wrapped up.

**Comment by Incoming5643** [comment]
Don't be too afraid of messing up this first one. You can always start fresh on a new PR if you need to

**Comment by CosmicScientist** [comment]
@Incoming5643 I'm hoping to figure out how to make them friendly to who spawned them. The holodeck can only spawn carp on the holodeck via the computer on a closed off side and some people are just too paranoid to go there or won't care. This lets it happen anywhere so long as you can deliver water to them.

@WJohn use it to soften up the target, to finish off the target, to attack them at the same time as the carp, as a distraction for a one person or a crowd, to frighten people into running maybe. It could be done in such a way to not tie you to the crime unlike a common revolver. And I could continue to look into fiddling with things until I can make a decent attack "dog".

@JJRcop I'm a tad confused, treat me as a child who knows nothing about Git or GitHub.

**Comment by dannno** [comment]
hilarious item great thinking :+1:

**Comment by Fox-McCloud** [comment]
Should be loyal to whoever created it.

**Comment by Jordie0608** [comment]
Set their faction var to the user's upon spawn to have them friendly to their creator.

**Comment by Tokiko1** [comment]
I think that you could include more than just 1 carp for 7 TC. Or lower the price.

**Comment by phil235** [comment]
I'd make it 2 or 3 TC to be honest, a single carp is pretty weak.

**Comment by CosmicScientist** [comment]
@Jordie0608 thank you I'll try to do that.

**Comment by JJRcop** [comment]
@CosmicScientist Sorry, I was just trying to shout as much git jargon at you as possible to make you concerned. None of what I said is sensible, just trying to give you a scare. :)

Jokes aside, this idea is pretty nice.

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (JJRcop) [inline code/game/objects/items/toys.dm]: Well it isn't compiling so I think this may be the culprit

**Comment by Chocobro** [comment]
big :+1:.

**Comment by CosmicScientist** [comment]
@JJRcop ;~; I went to bed afraid I did bad things. =P

**Comment by CosmicScientist** [comment]
I found time to work on this again however I'm at a loss as to how to change the carp's faction to the user's. Can anyone help? Is writing a proc to get/set faction unnecessary?

File: code/game/objects/items/dehy_carp.dm (hunk 1 above)
Comment (phil235) [inline code/game/objects/items/dehy_carp.dm]: yeah, using flick would be better.

**Comment by phil235** [comment]
@CosmicScientist 
`var/mob/living/simple_animal/hostile/carp/C = new /mob/living/simple_animal/hostile/carp(get_turf(src))`	
`C.faction |= "\ref[user]"`
(and give Swell() mob/user for argument)

**Comment by CosmicScientist** [comment]
@phil235 
Thank you.

**Comment by CosmicScientist** [comment]
@phil235 
Sorry for asking for more help but I got to try that out in the code and testing it tells me I did something wrong. I have the proc written like this:

dehy_carp/proc/Swell(mob/user)

Am I understanding what you mean as using it for the argument wrong? Is it because I always call Swell() without putting anything between the brackets?

**Comment by CosmicScientist** [comment]
By testing I mean putting up a test server and finding out the carp still wanted to murder me even though I created it through the syndicate uplink and then used a fire extinguisher on it.

**Comment by phil235** [comment]
Yeah, you need to put the mob in the brackets.

**Comment by CosmicScientist** [comment]
Thank you and just to make certain I'm not being a dolt.
When calling the proc I do:
Swell(mob/user)

The proc is written like this:
dehy_carp/proc/Swell(mob/user)

And as a hopefully final question, if this works for fire extinguishers/pouring water from a container, do they side with the faction of the mob who bought the carp from their uplink or the faction of the mob who put the water on them?

**Comment by CosmicScientist** [comment]
Since surely that'd give the mob who did the action? Because unless I wanted to do something such as "They imprint on the first thing they see so they don't kill that", I may need to get the user a different way.

**Comment by CosmicScientist** [comment]
How do I fix the merge branch piece? That was from I think originally doing git pull upstream master to make sure my code was up to date.

**Comment by Jordie0608** [comment]
To fix merge conflicts you need to pull from upstream master, then see what files were unable to be updated: those are the conflicting ones. If you're using a gui like tortoisegit you can go TortoiseGit > Resolve to see exactly which lines in each file are the trouble.

**Comment by JJRcop** [comment]
@CosmicScientist Don't be sorry about asking for help. If you didn't, life would be more difficult for all of us, and we're glad to share our git/BYOND wizardry to anyone who is willing to give up their title of "player" and trade it for "#coderbus conspirator!!!1111"

**Comment by CosmicScientist** [comment]
@Jordie0608 
Thanks, it was the main dmi that was conflicted and stopping everything else from merging... now it's telling me it's up-to-date but the build failed?

@JJRcop 
So long as you don't realise I'm actually a part of "#infiltrate coderbus" to take you down from within.

**Comment by JJRcop** [comment]
@CosmicScientist https://travis-ci.org/tgstation/-tg-station/jobs/58072286
You can see that there are three compile errors
```
code/datums/uplink_item.dm:380:error: /obj/item/toy/carpplushie/dehy_carp: undefined type path
code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm:292:error: /obj/item/toy/carpplushie/dehy_carp: undefined type path
code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm:294:error: dehy.Swell: undefined type: dehy.Swell
```

Now, I don't know why they are erroring, it looks fine to me from a glance, but that would be the problem.

**Comment by CosmicScientist** [comment]
@JJRcop 
I see now. Was I meant to add the tgstation.dme to the commit? Because I believe it's acting as if in Dream Maker the little check box beside the file isn't checked because my dehy_carp is in a new file itself due to how I didn't see it fitting elsewhere without being lost.

**Comment by JJRcop** [comment]
@CosmicScientist Yes, commit tgstation.dme. That is the file that stores all of the "check/unchecked" files. But first, go to Build > Preferences for tgstation, and make sure that the "FILE_DIRS" box is UNCHECKED

**Comment by CosmicScientist** [comment]
And now I just realised one more thing to add.

**Comment by CosmicScientist** [comment]
Well I think that's all done and dusted unless there was anything I missed, anything I could of done better or could also include.

**Pull request merged**